site/public/scripts/scroll-to-anchor.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * This script is used to scroll to anchors in the blog post page.
+ */
+
+/** @param {string} selector */
+const isElementLoaded = async selector => {
+  while (document.querySelector(selector) === null) {
+    await new Promise(resolve => requestAnimationFrame(resolve))
+  }
```

Comment:
how often does this poll? and did you try with throttling to 3G in chrome?

Reply:
don't know how often but it's the alternative to hardcoding wait time / interval & yes works in 3g